cypress/pageobject/Users/UserCreation.ts

```diff
@@ -1,45 +1,122 @@
 // UserCreation.ts
 export class UserCreationPage {
-  clickElementById(elementId: string) {
-    cy.get("#" + elementId).click();
+  private selectOptionContainingText(text: string) {
+    cy.get("[role='option']")
+      .contains(text)
+      .should("be.visible")
+      .click({ timeout: 10000 });
   }
 
-  typeIntoElementById(elementId: string, value: string) {
-    cy.get("#" + elementId)
-      .click()
-      .type(value);
+  clickProfileName() {
+    cy.get("#user-profile-name").click();
+  }
+  clickProfileButton() {
+    cy.get("#profile-button").click();
+  }
+  clickEditProfileButton() {
+    cy.get("#edit-cancel-profile-button").click();
+  }
+  clickAddUserButton() {
+    cy.get("#addUserButton").click();
+  }
+  clickSubmitButton() {
+    cy.get("#submit").click();
   }
 
-  typeIntoElementByIdPostClear(elementId: string, value: string) {
-    cy.get("#" + elementId)
-      .click()
-      .clear()
-      .click()
-      .type(value);
+  typeUserName(username: string) {
+    cy.get("#username").click().type(username);
   }
-  typeIntoElementByIdPostClearDob(elementId: string, value: string) {
-    cy.clickAndTypeDate("#" + elementId, value);
+  typeFirstName(firstName: string) {
+    cy.get("#firstName").click().type(firstName);
   }
-  clearIntoElementById(elementId: string) {
-    cy.get("#" + elementId)
-      .click()
-      .clear();
+  typeLastName(lastName: string) {
+    cy.get("#lastName").click().type(lastName);
+  }
+  typeEmail(email: string) {
+    cy.get("#email").click().type(email);
+  }
+
+  typePassword(password: string) {
+    cy.get("#password").click().type(password);
+  }
+  typePhoneNumber(phone_number: string) {
+    cy.get("#phoneNumber").click().type(phone_number);
+  }
+  typeConfirmPassword(password: string) {
+    cy.get("#c_password").click().type(password);
+  }
+  typeQualification(qualification: string) {
+    cy.get("#qualification").click().type(qualification);
+  }
+  typeDoctorExperience(experience: string) {
+    cy.get("#doctor_experience_commenced_on").click().type(experience);
+  }
+
+  typeDoctorMedicalCouncilRegNo(regNo: string) {
+    cy.get("#doctor_medical_council_registration").click().type(regNo);
+  }
+
+  typeWeeklyWorkingHours(workingHrs: string) {
+    cy.get("#weekly_working_hours").click().type(workingHrs);
+  }
+
+  typeAltPhoneNumber(emergencyPhoneNumber: string) {
+    cy.get("#altPhoneNumber").click().type(emergencyPhoneNumber);
   }
 
   typeIntoInputByName(inputName: string, value: string) {
```

Comment:
```suggestion

```

- no need of this, as you will be using reusable components for facility name selection